### PR TITLE
Pal 666 service by name only

### PR DESCRIPTION
**Pull request by nw1984**

### CODE REVIEW

File: resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java [hunk 1]
```diff
@@ -71,11 +69,12 @@ private LeafResource asFileResource(final File file) {
         if (i > 0) {
             extension = file.getName().substring(i + 1);
         }
-        URI dataServiceUri = clientConfiguration.getClientUri("data-service").orElseThrow(() -> new RuntimeException("Failed to find any instance of 'data-service'"));
+
+        String dataServiceName = "data-service";
```
Comment (ac74475) [inline resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java]: Use the static constant above instead

**Review by ac74475 — CHANGES_REQUESTED** [review]

File: resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ApplicationConfiguration.java [hunk 2]
```diff
@@ -17,11 +17,10 @@
 package uk.gov.gchq.palisade.service.resource.config;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.netflix.discovery.EurekaClient;
+import org.apache.kerby.kerberos.kerb.client.preauth.pkinit.ClientConfiguration;
```
Comment (dev-958) [inline resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ApplicationConfiguration.java]: !!! You what?

File: resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java [hunk 3]
```diff
@@ -39,10 +38,9 @@
 public class SimpleResourceService implements ResourceService {
     private static final Logger LOGGER = LoggerFactory.getLogger(SimpleResourceService.class);
 
-    private final ClientConfiguration clientConfiguration;
+    private static final String DATA_SERVICE_NAME = "data-service";
```
Comment (dev-958) [inline resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java]: Nope. This needs to be grabbed from a config file with the values duplicated in both k8s config and the eureka config, it's a reference to a data service that will handle a defined set of technologies/data servers so we may have a list of semantic handles for them.

**Review by ac74475 — DISMISSED** [review]

**Review by dev930018 — DISMISSED** [review]

**Review by dev-958 — CHANGES_REQUESTED** [review]

**Review by ac74475 — DISMISSED** [review]

File: resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ApplicationConfiguration.java [hunk 4]
```diff
@@ -78,49 +76,25 @@
...
+                .serviceName(serviceName);
         return () -> resourceConfig.getResources().stream()
                 .map(factory -> factory.build(connectionDetailMapper))
                 .collect(Collectors.toList());
     }
 
     /**
      * A container for a number of {@link StdResourcePrepopulationFactory} builders used for creating {@link uk.gov.gchq.palisade.resource.Resource}s
-     * These resources will be used for prepopulating the {@link ResourceService}
+     * These resources will be u fsed for prepopulating the {@link ResourceService}
```
Comment (developer01189998819991197253) [inline resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ApplicationConfiguration.java]:
```suggestion
     * These resources will be used for prepopulating the {@link ResourceService}
```

**Review by dev-958 — DISMISSED** [review]

File: resource-service/src/main/resources/application.yaml [hunk 5]
```diff
@@ -44,3 +44,4 @@ population:
 # Available options: simple, hadoop
 resource:
   implementation: simple
+
```
Comment (developer01189998819991197253) [inline resource-service/src/main/resources/application.yaml:47]: web.client.data-service: localhost:8082 missing from default application.yaml

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: resource-service/src/main/resources/application.yaml (hunk 5 above)
Comment (developer01189998819991197253) [reply resource-service/src/main/resources/application.yaml:47]: added

**Review by developer01189998819991197253 — DISMISSED** [review]

File: resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java [hunk 6]
```diff
@@ -77,6 +85,13 @@ private LeafResource asFileResource(final File file) {
                 .connectionDetail(new SimpleConnectionDetail().serviceName(this.dataServiceName));
     }
 
+    /**
+     * Query stream.
+     *
+     * @param uri  the uri
+     * @param pred the pred
+     * @return the stream
```
Comment (ac74475) [inline resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java]: can you be a bit more descriptive on these

**Review by ac74475 — CHANGES_REQUESTED** [review]

File: resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java (hunk 6 above)
Comment (developer01189998819991197253) [reply resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java]: updated description

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request closed without merging**